engine/src/test/groovy/graphql/nadel/e2e/nextgen/NadelE2ETest.groovy

```diff
@@ -458,7 +563,56 @@ class NadelE2ETest extends Specification {
         def response = nadel.execute(nadelExecutionInput)
 
         def executionResult = response.get()
-        assert calledService1
+
+        return [executionResult.getData(), executionResult.getErrors()]
+    }
+
+    Object[] testServices(
+            Map<String, String> overallSchema, // Map<ServiceName, OverallSchema>
+            Map<String, String> services, // Map<ServiceName, UnderlyingSchema>
+            Map<String, Map<Document, Map>> serviceCalls, // Map<ServiceName, Map<Query, Result>>
+            String query,
+            Map rawVariables = [:]
+    ) {
+        def astSorter = new AstSorter()
+        def makeServiceExecution = { String serviceName ->
+            println "On calling service '$serviceName'"
+            def calls = serviceCalls[serviceName]
+            def execution = Mock(ServiceExecution)
+            calls.each {
+                def expectedDocument = astSorter.sort(it.key)
+                println "Expecting query"
+                def result = new ServiceExecutionResult(it.value)
+                println printAstCompact(expectedDocument)
+                1 * execution.execute({ ServiceExecutionParameters sep ->
```

Comment:
I've noticed this `1 *` here doesn't actually enforce that it's been called once.

I think instead we should store these in a `Map<ServiceName, ServiceExecution>` and then run the `1 *` later in the `then:` to fix that. But not a big deal right now.